### ISSUE TITLE
Redesign Slack UI

### ISSUE THREAD

**Issue by nkantar** [nkantar/Microblot#112]
The initial idea of triggering everything through slash commands might not be optimal given Slack's fancy interactive capabilities. It would be a good idea to step back from the unquestioned CLI-like approach to reevaluate other options and potentially shift to a different approach.

**Comment by nkantar** [comment]
Main modal

![image](https://user-images.githubusercontent.com/568543/106362452-b9d27400-62d7-11eb-9da3-cbaf59e5c35e.png)

```json
{
	"type": "modal",
	"title": {
		"type": "plain_text",
		"text": "Microblot",
		"emoji": true
	},
	"close": {
		"type": "plain_text",
		"text": "Close",
		"emoji": true
	},
	"blocks": [
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Yak Overflow: <https://yakoverflow.microblot.io/|yakoverflow.microblot.io>\nYour page: <https://yakoverflow.microblot.io/authors/nik/|yakoverflow.microblot.io/authors/nik>\nPosts: 24 total, 3 yours\nCategories: 13"
			}
		},
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Your unpublished drafts",
				"emoji": true
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Django Site Dispatch, Improved"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Django Site Dispatch"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Introducing Lucy"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		},
		{
			"type": "actions",
			"elements": [
				{
					"type": "button",
					"text": {
						"type": "plain_text",
						"text": "New Post",
						"emoji": true
					},
					"value": "click_me_123",
					"action_id": "actionId-0"
				}
			]
		},
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Your published posts",
				"emoji": true
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Goals for 2021"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Python’s dict.setdefault Rocks"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Goals for 2020 in Review"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Hire Me! (v.2020.10)"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Introducing Autohook"
			},
			"accessory": {
				"type": "button",
				"text": {
					"type": "plain_text",
					"text": "Edit",
					"emoji": true
				},
				"value": "click_me_123",
				"action_id": "button-action"
			}
		}
	]
}
```

**Comment by nkantar** [comment]
# New/Edit post modal

To-do:

- [ ] Category field is a custom action one, use to populate help text below
- [ ] figure out draft saving

<img width="616" alt="image" src="https://user-images.githubusercontent.com/568543/106362918-4ed66c80-62da-11eb-9b0d-94115e28af35.png">

```json
{
	"type": "modal",
	"title": {
		"type": "plain_text",
		"text": "Microblot",
		"emoji": true
	},
	"submit": {
		"type": "plain_text",
		"text": "Publish",
		"emoji": true
	},
	"close": {
		"type": "plain_text",
		"text": "Cancel",
		"emoji": true
	},
	"blocks": [
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "New post"
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"action_id": "plain_text_input-action",
				"placeholder": {
					"type": "plain_text",
					"text": "What do you want to ask of the world?"
				}
			},
			"label": {
				"type": "plain_text",
				"text": "Title",
				"emoji": true
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"multiline": true,
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Body (Markdown)",
				"emoji": true
			}
		},
		{
			"dispatch_action": true,
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"dispatch_action_config": {
					"trigger_actions_on": [
						"on_character_entered"
					]
				},
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Category",
				"emoji": true
			}
		},
		{
			"type": "context",
			"elements": [
				{
					"type": "plain_text",
					"text": "Suggested category: python",
					"emoji": true
				}
			]
		}
	]
}
```

**Comment by nkantar** [comment]
Post modal buttons:

![IMG_0601](https://user-images.githubusercontent.com/568543/106373503-072e0000-632f-11eb-8387-105fd0d7af1a.jpg)

**Comment by nkantar** [comment]
# New post

<img width="612" alt="image" src="https://user-images.githubusercontent.com/568543/106373530-44928d80-632f-11eb-8a04-edac44078afb.png">

```json
{
	"type": "modal",
	"title": {
		"type": "plain_text",
		"text": "Microblot",
		"emoji": true
	},
	"submit": {
		"type": "plain_text",
		"text": "Publish",
		"emoji": true
	},
	"close": {
		"type": "plain_text",
		"text": "Cancel",
		"emoji": true
	},
	"blocks": [
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "New post"
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"action_id": "plain_text_input-action",
				"placeholder": {
					"type": "plain_text",
					"text": "What do you want to ask of the world?"
				}
			},
			"label": {
				"type": "plain_text",
				"text": "Title",
				"emoji": true
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"multiline": true,
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Body (Markdown)",
				"emoji": true
			}
		},
		{
			"dispatch_action": true,
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"dispatch_action_config": {
					"trigger_actions_on": [
						"on_character_entered"
					]
				},
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Category",
				"emoji": true
			}
		},
		{
			"type": "context",
			"elements": [
				{
					"type": "plain_text",
					"text": "Suggested category: python",
					"emoji": true
				}
			]
		},
		{
			"type": "actions",
			"elements": [
				{
					"type": "button",
					"text": {
						"type": "plain_text",
						"text": "Save draft",
						"emoji": true
					},
					"value": "click_me_123",
					"action_id": "actionId-0"
				}
			]
		}
	]
}
```

**Comment by nkantar** [comment]
# Edit draft

<img width="595" alt="image" src="https://user-images.githubusercontent.com/568543/106373575-c71b4d00-632f-11eb-8adb-4637ad52de5e.png">

```json
{
	"type": "modal",
	"title": {
		"type": "plain_text",
		"text": "Microblot",
		"emoji": true
	},
	"submit": {
		"type": "plain_text",
		"text": "Publish",
		"emoji": true
	},
	"close": {
		"type": "plain_text",
		"text": "Cancel",
		"emoji": true
	},
	"blocks": [
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Edit post draft"
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"action_id": "plain_text_input-action",
				"placeholder": {
					"type": "plain_text",
					"text": "What do you want to ask of the world?"
				}
			},
			"label": {
				"type": "plain_text",
				"text": "Title",
				"emoji": true
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"multiline": true,
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Body (Markdown)",
				"emoji": true
			}
		},
		{
			"dispatch_action": true,
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"dispatch_action_config": {
					"trigger_actions_on": [
						"on_character_entered"
					]
				},
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Category",
				"emoji": true
			}
		},
		{
			"type": "context",
			"elements": [
				{
					"type": "plain_text",
					"text": "Suggested category: python",
					"emoji": true
				}
			]
		},
		{
			"type": "actions",
			"elements": [
				{
					"type": "button",
					"text": {
						"type": "plain_text",
						"text": "Save draft :page_facing_up:",
						"emoji": true
					},
					"value": "click_me_123",
					"action_id": "actionId-0"
				}
			]
		},
		{
			"type": "actions",
			"elements": [
				{
					"type": "button",
					"text": {
						"type": "plain_text",
						"text": "Delete :wastebasket:",
						"emoji": true
					},
					"value": "click_me_123",
					"action_id": "actionId-0"
				}
			]
		}
	]
}
```

**Comment by nkantar** [comment]
# Edit published post

<img width="629" alt="image" src="https://user-images.githubusercontent.com/568543/106373600-09dd2500-6330-11eb-8fdd-ca7cd7059817.png">

```json
{
	"type": "modal",
	"title": {
		"type": "plain_text",
		"text": "Microblot",
		"emoji": true
	},
	"submit": {
		"type": "plain_text",
		"text": "Publish",
		"emoji": true
	},
	"close": {
		"type": "plain_text",
		"text": "Cancel",
		"emoji": true
	},
	"blocks": [
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Edit published post"
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"action_id": "plain_text_input-action",
				"placeholder": {
					"type": "plain_text",
					"text": "What do you want to ask of the world?"
				}
			},
			"label": {
				"type": "plain_text",
				"text": "Title",
				"emoji": true
			}
		},
		{
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"multiline": true,
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Body (Markdown)",
				"emoji": true
			}
		},
		{
			"dispatch_action": true,
			"type": "input",
			"element": {
				"type": "plain_text_input",
				"dispatch_action_config": {
					"trigger_actions_on": [
						"on_character_entered"
					]
				},
				"action_id": "plain_text_input-action"
			},
			"label": {
				"type": "plain_text",
				"text": "Category",
				"emoji": true
			}
		},
		{
			"type": "context",
			"elements": [
				{
					"type": "plain_text",
					"text": "Suggested category: python",
					"emoji": true
				}
			]
		},
		{
			"type": "actions",
			"elements": [
				{
					"type": "button",
					"text": {
						"type": "plain_text",
						"text": "Delete :no_entry_sign:",
						"emoji": true
					},
					"value": "click_me_123",
					"action_id": "actionId-0"
				}
			]
		}
	]
}
```

**Comment by nkantar** [comment]
# Post deletion confirmation

<img width="601" alt="image" src="https://user-images.githubusercontent.com/568543/106373673-bfa87380-6330-11eb-8cdd-614babe8cc9a.png">

```json
{
	"type": "modal",
	"title": {
		"type": "plain_text",
		"text": "Microblot",
		"emoji": true
	},
	"submit": {
		"type": "plain_text",
		"text": "Delete",
		"emoji": true
	},
	"close": {
		"type": "plain_text",
		"text": "Cancel",
		"emoji": true
	},
	"blocks": [
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Confirm post deletion?"
			}
		},
		{
			"type": "context",
			"elements": [
				{
					"type": "mrkdwn",
					"text": "This action is *irreversible!*"
				}
			]
		}
	]
}
```

**Comment by nkantar** [comment]
Implemented in bd9e3a141f7bb30832a04db2bacb2a4536d013b3.
